modules/rtmessageadd.php

```diff
@@ -147,10 +157,21 @@
 				$message['destination'] = $queue['email'];
 			$recipients = $message['destination'];
 
-			if($message['userid'] && $addmsg)
-				$message['mailfrom'] = $queue['email'] ? $queue['email'] : $user['email'];
-			if($message['userid'] && !$addmsg)
-				$message['mailfrom'] = $user['email'] ? $user['email'] : $queue['email'];
+			$helpdesk_sender_email = ConfigHelper::getConfig('phpui.helpdesk_sender_email');
+			if(!empty($helpdesk_sender_email)) {
+				$mailfrom = $helpdesk_sender_email;
+
+				if($mailfrom == 'queue')
+					$mailfrom = $queue['email'];
+				elseif($mailfrom == 'user')
+					$mailfrom = $user['email'];
+			} else {
+				if($message['userid'] && $addmsg)
+					$message['mailfrom'] = $queue['email'] ? $queue['email'] : $user['email'];
```

Comment:
these assignments should also be made to $mailfrom ?

Reply:
Thats a very good point, I will get to that in a moment.

While at it you explain me in detail what is the purpose of `phpui.helpdesk_backend_mode` in the mean time? The documentation is very limited on the subject.

I am trying to understand what are the pros/cons of using that.

Reply:
> these assignments should also be made to $mailfrom ?

@chilek, I am guessing that you mean `$addmsg` by that, and I don't believe that we need these.

These would again prefer user email as a `From` if set, the point of this PR is to force `From` regardless of other conditions.

Of course unless it has something to do with that `helpdesk_backend_mode` that I still hope that you would explain to me.